build_connectors.sh

```diff
@@ -1,4 +1,56 @@
-#!/bin/sh
+#!/usr/bin/env bash
+set -e
+
+REPO=https://github.com/Analog-Labs/chain-connectors
+REGISTRY_PATH=${REGISTRY_PATH:-analoglabs}
+DOCKER_IMAGE_NAME=base-ci-linux
+DOCKER_IMAGE_VERSION=latest
+CONNECTOR_IMAGE_VERSION=0.4.0
+
+# Check if docker is running
+if ! docker info > /dev/null 2>&1; then
+  echo "This script uses docker - please start docker and try again!"
+  exit 1
+fi
+
+# Build the base-ci-linux if necessary
+if [[ "$(docker images -q "${REGISTRY_PATH}/${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_VERSION}" 2> /dev/null)" == "" ]]; then
+  docker build \
+    -f ./ci/dockerfiles/base-ci-linux/Dockerfile \
+    --build-arg VCS_REF=$(git rev-parse HEAD) \
+    --build-arg BUILD_DATE=$(date +%Y%m%d) \
+    --no-cache \
+    -t "${REGISTRY_PATH}/${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_VERSION}" \
+    "./ci/dockerfiles/${DOCKER_IMAGE_NAME}"
+fi
+
+docker build \
+    -f ./ci/dockerfiles/builder/Dockerfile \
+    --no-cache \
+    -t "${REGISTRY_PATH}/builder:latest" \
+    .
```

Comment:
I'm using docker to reliably build all connector images, once I'm on M1 mac, I have to use `aarch64-unknown-linux-musl` instead of `x86_64-unknown-linux-musl`.

Building everything on docker fixes a lot of issues I had in my local environment.

Reply:
seems like your local setup is broken. the only thing that should be required for your M1 mac as you point out is changing the target triple. I welcome a change to detect the target triple based on the host arch.